### PR TITLE
Feature/2934 reaction bubble alignment and orientation

### PR DESCRIPTION
**Pull request by TGazica**
### 🎯 Goal

Add the ability to customise the alignment of the user reaction inside message options in Compose and XML sdks and adding the ability to customise the orientation of the bubble in the XML sdk. Add the ability to add vertical padding to `ViewRactionsView` and `EditReactionsView`.

### 🛠 Implementation details

Added `MessageOptionsUserReactionAlignment`  and `MessageOptionsUserReactionBubbleOrientation`. In Compose setting `messageOptionsUserReactionAlignment` inside `ChatTheme` will define the reaction alignment. Using `MessageListViewStyle.messageOptionsUserReactionAlignment` or `streamUiMessageOptionsUserReactionAlignment` will set the alignment and overriding `streamUiMessageOptionsUserReactionOrientation` will set the bubble orientation.

Added `verticalPadding` to `ViewRactionsView` and `EditReactionsView`. Also changed how padding is applied to 
`ViewRactionsView`.

### 🎨 UI Changes

| XML Before | XML After | Compose Before | Compose After |
| --- | --- | --- | --- |
| Screenshot_20220518_102951 | Screenshot_20220520_144418 | Screenshot_20220518_102507 | Screenshot_20220518_102543 |

### 🧪 Testing

<details>
  <summary>XML testing patch</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml b/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml
--- a/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml	(revision 8cd1829ac40c27e5826277b86de628e7aa6b18e6)
+++ b/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml	(date 1653050483422)
@@ -20,6 +20,7 @@
         android:layout_height="0dp"
         android:layout_marginHorizontal="0dp"
         android:clipToPadding="false"
+        app:streamUiMessageOptionsUserReactionAlignment="start"
         app:layout_constraintBottom_toTopOf="@+id/messageInputView"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
Index: stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml b/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml
--- a/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml	(revision 8cd1829ac40c27e5826277b86de628e7aa6b18e6)
+++ b/stream-chat-android-ui-components-sample/src/main/res/values/dimens.xml	(date 1653050596503)
@@ -10,4 +10,15 @@
 
     <dimen name="badge_horizontal_offset">8dp</dimen>
     <dimen name="badge_vertical_offset">6dp</dimen>
+
+    <dimen name="stream_ui_view_reactions_horizontal_padding">5dp</dimen>
+    <dimen name="stream_ui_view_reactions_vertical_padding">5dp</dimen>
+    <dimen name="stream_ui_view_reactions_item_size">20dp</dimen>
+    <dimen name="stream_ui_view_reactions_bubble_height">30dp</dimen>
+
+    <dimen name="stream_ui_edit_reactions_horizontal_padding">10dp</dimen>
+    <dimen name="stream_ui_edit_reactions_vertical_padding">5dp</dimen>
+    <dimen name="stream_ui_edit_reactions_item_size">30dp</dimen>
+    <dimen name="stream_ui_edit_reactions_bubble_height">40dp</dimen>
+
 </resources>
Index: stream-chat-android-ui-components-sample/src/main/res/values-v29/themes.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/values-v29/themes.xml b/stream-chat-android-ui-components-sample/src/main/res/values-v29/themes.xml
--- a/stream-chat-android-ui-components-sample/src/main/res/values-v29/themes.xml	(revision 8cd1829ac40c27e5826277b86de628e7aa6b18e6)
+++ b/stream-chat-android-ui-components-sample/src/main/res/values-v29/themes.xml	(date 1653050596499)
@@ -12,6 +12,7 @@
 
     <style name="StreamTheme" parent="@style/StreamUiTheme">
         <item name="streamUiChannelActionsDialogStyle">@style/ChannelActionsDialogTheme</item>
+        <item name="streamUiSingleReactionBubbleOrientation">start</item>
     </style>
 
     <style name="ChannelActionsDialogTheme" parent="StreamUi.ChannelList.ActionsDialog">

```

</details>

<details>
  <summary>Compose testing patch</summary>

```
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt	(revision 14ae19d5c672c5e70f6d7416d522dbcbc64f5118)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt	(date 1653379823213)
@@ -49,6 +49,7 @@
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.common.MessageOptionsUserReactionAlignment
 import io.getstream.chat.android.common.state.DeletedMessageVisibility
 import io.getstream.chat.android.common.state.MessageMode
 import io.getstream.chat.android.common.state.Reply
@@ -93,7 +94,7 @@
         val channelId = intent.getStringExtra(KEY_CHANNEL_ID) ?: return
 
         setContent {
-            ChatTheme(dateFormatter = ChatApp.dateFormatter) {
+            ChatTheme(dateFormatter = ChatApp.dateFormatter, messageOptionsUserReactionAlignment = MessageOptionsUserReactionAlignment.START) {
                 MessagesScreen(
                     channelId = channelId,
                     onBackPressed = { finish() },
```

</details>

XML:

1. Build the app on this brach and take a look at how the message options user reactions are aligned, oriented and padding and size of the reactions
2. Apply the XML patch above
3. Repeat step one

Compose:

1. Build the app and take a look at how the message options user reactions are aligned
2. Apply the Compose patch above
3. Repeat step one

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

ezgif-5-a21781cf0f

### CODE REVIEW

File: stream-chat-android-ui-common/detekt-baseline.xml [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <ID>ComplexMethod:MessageFooterVisibility.kt$ @InternalStreamChatApi public fun MessageFooterVisibility.shouldShowMessageFooter( message: Message, isLastMessageInGroup: Boolean, nextMessage: Message?, ): Boolean</ID>
     <ID>LongMethod:MessageListItemLiveData.kt$MessageListItemLiveData$ @Suppress("ComplexMethod") private fun groupMessages(messages: List&lt;Message>?, currentUserId: String): List&lt;MessageListItem></ID>
     <ID>LongParameterList:Linkify.kt$Linkify$( links: MutableList&lt;LinkSpec>, s: Spannable, pattern: Pattern, schemes: Array&lt;String>, matchFilter: Linkify.MatchFilter?, transformFilter: Linkify.TransformFilter?, )</ID>
+    <ID>MagicNumber:MessageOptionsUserReactionAlignment.kt$MessageOptionsUserReactionAlignment.BY_USER_INVERTED$3</ID>
```
Comment (JcMinarro) [inline stream-chat-android-ui-common/detekt-baseline.xml:8]: Can we fix this MagicNumber?

Comment (TGazica) [reply stream-chat-android-ui-common/detekt-baseline.xml:8]: The thing is it is not a MagicNumber, I would say this is a false positive, take a look in `MessageOptionsUserReactionAlignment` that is in this pr. This is reported for a single enum value in this enum and i don't understand why.

Comment (filbabic) [reply stream-chat-android-ui-common/detekt-baseline.xml:8]: getrekt strikes again 😆

**Review by MarinTolic — CHANGES_REQUESTED** [review]
Requested changes due to kdocs comments, other than that the feature is looking fine. :smile: 

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt [hunk 2]
```diff
@@ -249,4 +259,9 @@ public object ChatTheme {
         @Composable
         @ReadOnlyComposable
         get() = LocalMessageAlignmentProvider.current
+
+    public val messageOptionsUserReactionAlignment: MessageOptionsUserReactionAlignment
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt:263]: Missing kdocs here

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/view/ViewReactionsViewStyle.kt [hunk 3]
```diff
@@ -88,6 +94,8 @@ public data class ViewReactionsViewStyle(
             private var bubbleBorderWidthMine: Float = DEFAULT_BUBBLE_BORDER_WIDTH_MINE
             @Px
             private var bubbleBorderWidthTheirs: Float? = null
+            private var messageOptionsUSerReactionOrientation: Int =
```
[the remark this answers is not in the document]
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/view/ViewReactionsViewStyle.kt]: refactored.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/view/ViewReactionsViewStyle.kt [hunk 4]
```diff
@@ -49,6 +50,8 @@ public data class ViewReactionsViewStyle(
     @Px public val smallTailBubbleCy: Int,
     @Px public val smallTailBubbleRadius: Int,
     @Px public val smallTailBubbleOffset: Int,
+    @Px public val verticalPadding: Int,
```
[the remark this answers is not in the document]
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/view/ViewReactionsViewStyle.kt:78]: kdocs added

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/edit/EditReactionsViewStyle.kt [hunk 5]
```diff
@@ -42,6 +42,7 @@ public data class EditReactionsViewStyle(
     @Px public val smallTailBubbleRadius: Int,
     @Px public val smallTailBubbleOffset: Int,
     public val reactionsColumn: Int,
+    @Px public val verticalPadding: Int
```
[the remark this answers is not in the document]
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/reactions/edit/EditReactionsViewStyle.kt:65]: kdocs added

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt (hunk 2 above)
Comment (TGazica) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt:266]: kdocs added

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**